apps/chatbots/tests/test_sessions_table_view.py

```diff
@@ -0,0 +1,256 @@
+        until get_table_data() is called.
+        """
+        data = chatbot_with_sessions
+        view = create_view_instance(data["team"], data["user"], data["chatbot"])
+
+        # Get the base queryset (used for counting)
+        queryset = view.get_queryset()
+
+        # The most expensive annotations should not be present yet
+        query_annotations = queryset.query.annotations
+        # experiment_versions is the slowest annotation - it should not be in the base queryset
+        assert "experiment_versions" not in query_annotations
+        # message_count should also not be in the base queryset
+        assert "message_count" not in query_annotations
+        # Note: last_message_created_at may be present due to filter requirements, which is acceptable
+
+    def test_pagination_works_with_annotations(self, chatbot_with_sessions):
+        """Test that pagination works correctly with deferred annotations."""
+        data = chatbot_with_sessions
+        team = data["team"]
+        user = data["user"]
+        chatbot = data["chatbot"]
+
+        # Request with pagination (page size 2)
+        url = reverse("chatbots:sessions-list", kwargs={"team_slug": team.slug, "experiment_id": chatbot.id})
+        request = create_authenticated_request(team, user, url, {"per_page": "2"})
+
+        view = ChatbotSessionsTableView.as_view()
+        response = view(request, team_slug=team.slug, experiment_id=chatbot.id)
+
+        assert response.status_code == 200
+        table = response.context_data["table"]
+
+        # Verify pagination is active
+        assert table.paginator is not None
+        assert table.paginator.num_pages > 1
+
+        # Verify the displayed rows have annotations
+        # Access the actual record from BoundRow using row.record
+        for row in table.page:
+            record = row.record
+            assert hasattr(record, "message_count")
+            assert hasattr(record, "last_message_created_at")
+            assert hasattr(record, "experiment_versions")
+
+    def test_empty_chatbot_renders_correctly(self, db):
+        """Test that a chatbot with no sessions renders correctly."""
+        team_with_users = TeamWithUsersFactory.create()
+        user = team_with_users.members.first()
+        chatbot = ChatbotFactory(team=team_with_users, owner=user)
+
+        url = reverse("chatbots:sessions-list", kwargs={"team_slug": team_with_users.slug, "experiment_id": chatbot.id})
+        request = create_authenticated_request(team_with_users, user, url)
+
+        view = ChatbotSessionsTableView.as_view()
+        response = view(request, team_slug=team_with_users.slug, experiment_id=chatbot.id)
+
+        assert response.status_code == 200
+        assert isinstance(response.context_data["table"], ChatbotSessionsTable)
+        assert len(response.context_data["table"].data) == 0
+
+    def test_session_without_messages_has_null_annotations(self, db):
```

Comment:
Does it feel weird to test this?

Reply:
It's an edge case test to ensure it doesn't error and that the join is working correctly.